applications/CoSimulationApplication/python_scripts/data_transfer_operators/kratos_mapping_with_debug.py

```diff
@@ -0,0 +1,379 @@
+
+        # Create a VTK output process object with the provided settings and the current model
+        pre_process_origin = vtk_output_process.Factory(pre_origin_settings, model_origin)
+        # Initialize and execute the VTK output process
+        pre_process_origin.ExecuteInitialize()
+        pre_process_origin.ExecuteInitializeSolutionStep()
+
+        # Set the output paths
+        pre_destination_settings = copy.deepcopy(destination_settings)
+        path_splitted = Path(destination_output_path).parts
+        directory_name = Path(*path_splitted[:-1])  # All but the last part
+        base_name = path_splitted[-1]  # Last part
+        if self.auxiliary_debug_counter:
+            directory_name /= name_prefix  # Append the name prefix
+        file_path = directory_name / f"pre_map_{base_name}"
+        pre_destination_settings["Parameters"]["output_path"].SetString(str(file_path))
+
+        # Create a VTK output process object with the provided settings and the current model
+        pre_process_destination = vtk_output_process.Factory(pre_destination_settings, model_destination)
+        # Initialize and execute the VTK output process
+        pre_process_destination.ExecuteInitialize()
+        pre_process_destination.ExecuteInitializeSolutionStep()
+
+        # Store the VTK output processes for debugging
+        self.__debug_vtk_pre[current_identifier][variable_identifier_tuple] = (pre_process_origin, pre_process_destination)
+
+        # Set the output paths
+        post_origin_settings = copy.deepcopy(origin_settings)
+        path_splitted = Path(origin_output_path).parts
+        directory_name = Path(*path_splitted[:-1])  # All but the last part
+        base_name = path_splitted[-1]  # Last part
+        if self.auxiliary_debug_counter:
+            directory_name /= name_prefix  # Append the name prefix
+        file_path = directory_name / f"post_map_{base_name}"
+        post_origin_settings["Parameters"]["output_path"].SetString(str(file_path))
+
+        # Create a VTK output process object with the provided settings and the current model
+        post_process_origin = vtk_output_process.Factory(post_origin_settings, model_origin)
+        # Initialize and execute the VTK output process
+        post_process_origin.ExecuteInitialize()
+        post_process_origin.ExecuteInitializeSolutionStep()
+
+        # Set the output paths
+        post_destination_settings = copy.deepcopy(destination_settings)
+        path_splitted = Path(destination_output_path).parts
+        directory_name = Path(*path_splitted[:-1])  # All but the last part
+        base_name = path_splitted[-1]  # Last part
+        if self.auxiliary_debug_counter:
+            directory_name /= name_prefix  # Append the name prefix
+        file_path = directory_name / f"post_map_{base_name}"
+        post_destination_settings["Parameters"]["output_path"].SetString(str(file_path))
```

Comment:
is the `str` needed? I was quite sure that pybind does the conversion automatically

Reply:
In give issues in Windows (at least my machine)

Reply:
Fair enough 

Can you post the error please, for reference?

Reply:
TypeError: SetString(): incompatible function arguments. The following argument types are supported:
    1. (self: Kratos.Parameters, arg0: str) -> None